utils/build-script-impl

```diff
@@ -3443,7 +3443,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 if [ "${BUILD_LIBPARSER_ONLY}" ]; then
                     # We don't have a toolchain so we should install to the specified dir
                     DYLIB_DIR="${INSTALL_DESTDIR}"
-                    MODULE_DIR="${INSTALL_DESTDIR}"
+                    MODULE_DIR="${INSTALL_DESTDIR}/SwiftSyntax.swiftmodule/${SWIFT_HOST_VARIANT_ARCH}"
```

Comment:
Can we use `${product}` for `SwiftSyntax` here?

Reply:
Good idea. Changed it.